doc/man1/openssl-s_client.pod.in

```diff
@@ -1007,6 +1007,27 @@ option: any verify errors are then returned aborting the handshake.
 The B<-bind> option may be useful if the server or a firewall requires
 connections to come from some particular address and or port.
 
+=head2 Note on Non-Interactive Use
+
+When B<s_client> is run in a non-interactive environment (e.g., a cron job or
+a script without a valid I<stdin>), it may close the connection before
+receiving all the necessary TLS session data, especially with TLS 1.3.
+To prevent this, consider using the B<-ign_eof> option to keep B<s_client>
+running even after reaching EOF from I<stdin>.
+
+Additionally, redirecting input explicitly, such as by using B<E<lt>/dev/null>,
+can ensure the proper handling of I<stdin> and avoid premature connection
+closure.
+
+For example:
+
+ openssl s_client -connect <server address>:443 -tls1_3
+                  -sess_out /path/to/tls_session_params_file
+                  -ign_eof </dev/null
+
+This ensures that the client remains active until all session data is
+properly received and the server closes the connection.
```

Comment:
What happens when the server keeps the connection open, waiting for the client to close first? This is not particularly unlikely, for example:

```
$ openssl s_client -brief -ign_eof -starttls smtp -connect localhost:25 < /dev/null
Can't use SSL_get_servername
CONNECTION ESTABLISHED
Protocol version: TLSv1.3
Ciphersuite: TLS_AES_256_GCM_SHA384
Peer certificate: CN = [...]
Hash used: SHA256
Signature type: RSA-PSS
Verification: OK
Server Temp Key: X25519, 253 bits
250 CHUNKING
<long pause>
```
What I typically do instead is arrange for a graceful application-level disconnect, for example:
```
$ printf 'QUIT\r\n' | openssl s_client -brief -ign_eof -starttls smtp -connect $(uname -n):25
CONNECTION ESTABLISHED
Protocol version: TLSv1.3
Ciphersuite: TLS_AES_256_GCM_SHA384
Peer certificate: CN = [...]
Hash used: SHA256
Signature type: RSA-PSS
Verification: OK
Server Temp Key: X25519, 253 bits
250 CHUNKING
221 2.0.0 Bye
```

For HTTP/1.1, one might send a request with a 'Connection: close' header, ... I am not sure that '-ign_eof' is unconditionally a sound alternative. Similar hang with HTTPS:
```
$ </dev/null openssl s_client -brief -ign_eof -connect 127.0.0.1:443
Can't use SSL_get_servername
CONNECTION ESTABLISHED
Protocol version: TLSv1.3
Ciphersuite: TLS_AES_256_GCM_SHA384
Peer certificate: CN = chapelte.ch
Hash used: SHA512
Signature type: RSA-PSS
Verification: OK
Server Temp Key: X25519, 253 bits
[long pause]
^C
```

Reply:
What additions would you recommend for the documentation?

Reply:
I have updated the documentation to address the potential for hangs when using `-ign_eof` in non-interactive environments. The revised section now includes a detailed explanation of these scenarios, along with examples inspired by your suggestions to demonstrate how to avoid such issues with application-specific disconnect commands.